packages/astro/components/ViewTransitions.astro

```diff
@@ -219,13 +219,13 @@ const { fallback = 'animate' } = Astro.props as Props;
 				for (const s2 of newDocument.scripts) {
 					if (
 						// Inline
-						(s1.textContent && s1.textContent === s2.textContent) ||
+						(!s1.src && s1.textContent === s2.textContent) ||
 						// External
-						(s1.type === s2.type && s1.src === s2.src)
+						(s1.src && s1.type === s2.type && s1.src === s2.src)
 					) {
-						s2.remove();
-					} else {
-						s1.remove();
+						// the old script is in the new document: we mark it as executed to prevent re-execution
+						s2.dataset.astroExec = '';
```

Comment:
very clever, this also means they will exist on the next page transition, so they won't be removed there either.